tuta-sdk/rust/sdk/src/date/calendar_facade.rs

```diff
@@ -167,14 +168,24 @@ impl CalendarFacade {
 			});
 		};
 
-		let (timestamp_start, timestamp_end) = match self.parse_date_to_all_day_range(date, offset)
-		{
+		let (start_range, _) = match self.parse_date_to_all_day_range(start_date, offset) {
+			Ok(value) => value,
+			Err(err) => return Err(err),
+		};
```

Comment:
```suggestion
		let (start_range, _) = self.parse_date_to_all_day_range(start_date, offset)?;
```

and same for the other one